lib/command/setup_app.rb

```diff
@@ -26,22 +29,69 @@ def call # rubocop:disable Metrics/MethodLength
               "or run 'cpl apply-template #{templates.join(' ')} -a #{config.app}'."
       end
 
+      create_secret_and_policy_if_not_exist unless config.options[:skip_secret_access_binding]
+
       Cpl::Cli.start(["apply-template", *templates, "-a", config.app])
 
-      return if config.options[:skip_secret_access_binding]
+      bind_identity_to_policy unless config.options[:skip_secret_access_binding]
+    end
+
+    private
+
+    def create_secret_and_policy_if_not_exist
+      create_secret_if_not_exists
+      create_policy_if_not_exists
+
+      progress.puts
+    end
+
+    def create_secret_if_not_exists
+      if cp.fetch_secret(app_secrets)
+        progress.puts("Secret '#{app_secrets}' already exists. Skipping creation...")
+      else
+        step("Creating secret '#{app_secrets}'") do
+          cp.apply_hash(build_secret_hash)
+        end
+      end
+    end
+
+    def create_policy_if_not_exists
+      if cp.fetch_policy(app_secrets_policy)
+        progress.puts("Policy '#{app_secrets_policy}' already exists. Skipping creation...")
+      else
+        step("Creating policy '#{app_secrets_policy}'") do
+          cp.apply_hash(build_policy_hash)
+        end
+      end
+    end
+
+    def build_secret_hash
+      {
+        "kind" => "secret",
+        "name" => app_secrets,
+        "type" => "dictionary",
+        "data" => {}
+      }
+    end
+
+    def build_policy_hash
+      {
+        "kind" => "policy",
+        "name" => app_secrets_policy,
+        "targetKind" => "secret",
+        "targetLinks" => ["//secret/#{app_secrets}"]
+      }
+    end
 
+    def bind_identity_to_policy
       progress.puts
 
-      if cp.fetch_identity(app_identity).nil? || cp.fetch_policy(app_secrets_policy).nil?
-        raise "Can't bind identity to policy: identity '#{app_identity}' or " \
-              "policy '#{app_secrets_policy}' doesn't exist. " \
-              "Please create them or use `--skip-secret-access-binding` to ignore this message." \
-              "You can also set a custom secrets name with `secrets_name` " \
-              "and a custom secrets policy name with `secrets_policy_name` " \
-              "in the `.controlplane/controlplane.yml` file."
+      if cp.fetch_identity(app_identity).nil?
```

Comment:
It's now impossible for the policy not to exist here, as `create_policy_if_not_exists` will have been called earlier.